R-packages/covidcast/R/covidcast.R

```diff
@@ -529,58 +532,59 @@ covidcast_days <- function(data_source, signal, start_day, end_day, geo_type,
   for (i in seq(ndays + 1)) {
     query_day <- start_day + i - 1
     day_str <- date_to_string(query_day)
-    dat[[i]] <- covidcast(data_source = data_source,
-                          signal = signal,
-                          time_type = "day",
-                          geo_type = geo_type,
-                          time_values = day_str,
-                          geo_value = geo_value,
-                          as_of = as_of,
-                          issues = issues,
-                          lag = lag)
+    res <- covidcast(data_source = data_source,
+                     signal = signal,
+                     time_type = "day",
+                     geo_type = geo_type,
+                     time_values = day_str,
+                     geo_value = geo_value,
+                     as_of = as_of,
+                     issues = issues,
+                     lag = lag)
+
+    if (is.null(res)) {
+      warn(paste0("Fetching ", signal, " from ", data_source, " for ",
+                  query_day, " in geography '", geo_value, "': no results"),
+           data_source = data_source,
+           signal = signal,
+           day = query_day,
+           geo_value = geo_value,
+           class = "covidcast_fetch_failed")
+
+      next
+    }
+
+    dat[[i]] <- res
+
     summary <- sprintf(
-      "Fetched day %s: %s, %s, num_entries = %s",
+      "Fetched day %s: num_entries = %s",
```

Comment:
Does the change to this line represent a change to the semantics of the output or is it just a change in compatibility between the two object types?  That is to say, the old `dat[[i]]` has attributes `result`, `message`, and `epidata`--does the new `dat[[i]]` have something different?

Reply:
Good question, and perhaps we should document this more clearly in a comment. The old format would return error codes in the JSON. The CSV format does not have a field for error messages; instead it uses HTTP status codes, so the 

```r
 httr::stop_for_status(response, task = "fetch data from API")
```

line in `.request` will do the main error reporting if we don't get HTTP 200 back.